### DISCUSSION TITLE
Possible to show output of process on page?

### DISCUSSION THREAD

**Discussion by parterburn** [Shopify/maintenance_tasks#535] [Ideas]
Awesome video walkthrough, @adrianna-chang-shopify! In the example, you have `puts element` in the process method, which got me thinking this gem could be a nice drop-in to run some ad-hoc reporting for certain use-cases...if the output of those puts was displayed on the page somewhere.

Is that possible with this gem setup by chance?

**Comment by adrianna-chang-shopify** [comment]
Hey @parterburn !  Can you expand on your use case a bit more for me? The maintenance tasks gem is ultimately designed to make it easy to perform one-off data changes in production. I don't know if it makes sense to use it as a reporting tool, although you're welcome to write a task that outputs to your application's Rails logger or emits metrics to an observability service like DataDog.

**Comment by internethostage** [comment]
I was also looking into having some observability on the output of a task.
My use case is basically that I have a task that I run every once in a while to import data from a csv file.
Its useful for me to see how many records were created and how many were updated from this task.


As suggested above, I could just send this data to an external service like Datadog, or go search on the Rails logger, but I am stubborn and wanted it on the task ui. 😄

I did a hack to achieve this, but its not pretty.
I patched the following 2 methods:

```ruby
    def persist_progress(number_of_ticks, duration)
      update_columns(arguments: arguments.merge("task_results" => task.task_results)) if task.respond_to?(:task_results)

      super
    end

    def start(count)
      task.define_singleton_method(:task_results) do
        @task_results ||= Hash.new(0)
      end
      task.task_results.merge!(arguments["task_results"]) if arguments["task_results"]

      super
    end
```

Then on the task itself whenever I want to record something I have the `task_results` hash to dump what I need. This gets persisted to the arguments column of the task on each run of the process method, so it even works if the task is paused.
The `task_results` is displayed on the webui under the `Arguments` section of an executed run.

If anyone else believes they could benefit from this, I can give this idea an extra spin with a cleaner approach.

**Reply by artyrcheek** [reply]
Yeah, I would like this, Just a simple output log would go a long way for us.
